playground-common/src/Control/Monad/Web.hs

```diff
@@ -22,18 +25,25 @@ class Monad m =>
   doRequest :: Manager -> Request -> m (Either Text (Response LBS.ByteString))
 
 instance MonadWeb IO where
-  makeManager = newManager $ tlsManagerSettings {managerModifyRequest = pure . addUserAgent}
+  makeManager = newManager $ tlsManagerSettings
+    { managerModifyRequest = pure . addUserAgent
+    , managerResponseTimeout = maybe
+      (managerResponseTimeout tlsManagerSettings)
+      responseTimeoutMicro . toIntegralSized
+      $ toMicroseconds maxInterpretationTime }
   doRequest manager request = Right <$> httpLbs request manager
 
 instance MonadWeb m => MonadWeb (LoggingT m) where
   makeManager = lift makeManager
   doRequest manager = lift . doRequest manager
 
-
 instance MonadWeb m => MonadWeb (ExceptT e m) where
   makeManager = lift makeManager
   doRequest manager = lift . doRequest manager
 
 instance MonadWeb m => MonadWeb (ReaderT a m) where
   makeManager = lift makeManager
   doRequest manager = lift . doRequest manager
+
+maxInterpretationTime :: Second
+maxInterpretationTime = 160
```

Comment:
Probably it's make sense to keep it at the `MonadReader` environment, and read for each app from the shell environment/configuration file/application arguments.